### PR TITLE
feat: sklearn_pipe property to pipelines

### PR DESCRIPTION
**Pull request by MartinBernstorff**

### CODE REVIEW

**Comment by MartinBernstorff** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Aarhus-Psychiatry-Research/psycop-common/732?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #729** <a href="https://app.graphite.dev/github/pr/Aarhus-Psychiatry-Research/psycop-common/729?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #727** <a href="https://app.graphite.dev/github/pr/Aarhus-Psychiatry-Research/psycop-common/727?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #731** <a href="https://app.graphite.dev/github/pr/Aarhus-Psychiatry-Research/psycop-common/731?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #732** <a href="https://app.graphite.dev/github/pr/Aarhus-Psychiatry-Research/psycop-common/732?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/Aarhus-Psychiatry-Research/psycop-common/732?utm_source=stack-comment).

File: psycop/common/model_training_v2/test_pipeline.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def test_v2_train_model_pipeline(tmpdir: Path):
                 AgeFilter(min_age=4, max_age=99, age_col_name="pred_age"),
             ),
             task=BinaryClassificationTask(
-                task_pipe=BinaryClassificationPipeline(
+                pipe=BinaryClassificationPipeline(
```
Comment (HLasse) [inline psycop/common/model_training_v2/test_pipeline.py:64]: Not sure I agree with the renaming here. In my mind, `pipe` by default refers to an sklearn pipe, and we're making a distinction between the two here by calling them `task_pipe` and `sklearn_pipe`

File: psycop/common/model_training_v2/trainer/task/binary_classification/binary_classification_task.py [hunk 2]
```diff
@@ -20,12 +22,9 @@ def polarsframe_to_series(polarsframe: PolarsFrame) -> pl.Series:
...
 
 @BaselineRegistry.tasks.register("binary_classification")
+@dataclass(frozen=True)
 class BinaryClassificationTask(BaselineTask):
-    def __init__(
-        self,
-        task_pipe: BinaryClassificationPipeline,
-    ):
-        self.pipe = task_pipe
+    pipe: BinaryClassificationPipeline
```
Comment (HLasse) [inline psycop/common/model_training_v2/trainer/task/binary_classification/binary_classification_task.py:27]: same comment as previously

**Review by HLasse — APPROVED** [review]

**Comment by MartinBernstorff** [comment]
### Merge activity

* **Jan 18, 3:37 PM**: @MartinBernstorff merged this pull request with [Graphite](https://app.graphite.dev/github/pr/Aarhus-Psychiatry-Research/psycop-common/732).

**Pull request merged**